spec/definitions.yaml

```
@@ -1019,6 +1019,24 @@ definitions:
             type: string
             description: |
               Azure VM size to use for all nodes in the node pool. _(Validated against available VM sizes.)_
+          spot_instances:
+            description: |
+              Attributes defining the presence and configuration of Spot Instances.
+            type: object
+            properties:
+              enabled:
+                description: |
+                  Whether the feature is enabled or not.
+                type: boolean
+                default: false
+              max_price:
+                description: |
+                  The maximum price that a single node pool VM instance can reach before it is deallocated.
```

I suggest to add info on the currency for clarity

```suggestion
                  The maximum price in USD that a single node pool VM instance can reach before it is deallocated.
```